File: VB/Views/MainView.vb
Imports DevExpress.XtraEditors
Imports TodoApp.ViewModels

Namespace TodoApp
    Public Partial Class MainView
        Inherits XtraForm
        Public Sub New()
            InitializeComponent()
            ' Initializing bindings only at runtime  
            If Not mvvmContext.IsDesignMode Then
                InitializeNavigation()
                InitializeBindings()
            End If
        End Sub
        Sub InitializeNavigation()
            ' creating the NavigationFrame as INavigationService
            Dim navigationService = DevExpress.Utils.MVVM.Services.NavigationService.Create(navigationFrame)
            ' registering the service instance
            mvvmContext.RegisterService(navigationService)
            ' Initialize the Fluent API
            Dim fluent = mvvmContext.OfType(Of AppViewModel)()
            ' Bind the OnShown command to the Shown event
#Disable Warning
            fluent.WithEvent(Me, "Shown").EventToCommand(Sub(x) x.OnShown())
#Enable Warning Warning
        End Sub
        Sub InitializeBindings()
            ' Initialize the Fluent API
            Dim fluent = mvvmContext.OfType(Of AppViewModel)()
            ' Bind the Title property to the Text
            fluent.SetBinding(Me, Function(view) view.Text, Function(x) x.Title)
        End Sub
    End Class
End Namespace

File: CS/Views/MainView.cs
namespace TodoApp {
    using DevExpress.Utils.MVVM.Services;
    using DevExpress.XtraEditors;
    using TodoApp.ViewModels;

    public partial class MainView : XtraForm {
        public MainView() {
            InitializeComponent();
            // Initializing bindings only at runtime  
            if(!mvvmContext.IsDesignMode) {
                InitializeNavigation();
                InitializeBindings();
            }
        }
        void InitializeNavigation() {
            // creating the NavigationFrame as INavigationService
            var navigationService = NavigationService.Create(navigationFrame);
            // registering the service instance
            mvvmContext.RegisterService(navigationService);
            // Initialize the Fluent API
            var fluent = mvvmContext.OfType<AppViewModel>();
            // Bind the OnShown command to the Shown event
            fluent.WithEvent(this, "Shown")
                .EventToCommand(x => x.OnShown);
        }
        void InitializeBindings() {
            // Initialize the Fluent API
            var fluent = mvvmContext.OfType<AppViewModel>();
            // Bind the Title property to the Text
            fluent.SetBinding(this, view => view.Text, x => x.Title);
        }
    }
}